src/mesh/p4est_mesh.jl

```diff
@@ -430,32 +697,30 @@ function refine_fn(p4est, which_tree, quadrant)
   end
 end
 
+cfunction(::Val{refine_fn}, ::Val{2}) = @cfunction(refine_fn, Cint, (Ptr{p4est_t}, Ptr{p4est_topidx_t}, Ptr{p4est_quadrant_t}))
+cfunction(::Val{refine_fn}, ::Val{3}) = @cfunction(refine_fn, Cint, (Ptr{p8est_t}, Ptr{p4est_topidx_t}, Ptr{p8est_quadrant_t}))
+
 # Refine marked cells and rebalance forest.
 # Return a list of all cells that have been refined during refinement or rebalancing.
 function refine!(mesh::P4estMesh)
   # Copy original element IDs to quad user data storage
   original_n_cells = ncells(mesh)
   save_original_ids(mesh)
 
-  init_fn_c = @cfunction(init_fn, Cvoid,
-    (Ptr{p4est_t}, Ptr{p4est_topidx_t}, Ptr{p4est_quadrant_t}))
+  init_fn_c = cfunction(Val(init_fn), Val(ndims(mesh)))
+  refine_fn_c = cfunction(Val(refine_fn), Val(ndims(mesh)))
```

Comment:
This is the best idea I had to prevent duplicate code that dispatches by mesh dimension, only to call `@cfunction` once with `p4est` types and once with `p8est` types.
Any better ideas?

Reply:
Might be okay, I guess. 
Minor comment: `init_fn` is a function, so you don't need a `Val` here but can use `typeof(init_fn)`.

Reply:
Yes, that works. Thanks!